### PR TITLE
Assortment Pull Request #3, "wow sdk 2013 is so useful" edition

### PR DESCRIPTION
**Pull request by Adrthegamedev**
I did more digging through SDK 2013, copying a list of every datadesc via VS Code. This PR is mostly from that, although I did grab some stuff from other sources (such as in-engine datadesc dumps, or from stuff I noticed was on VDC but not here).

**Changes:**

`BaseHelicopter`:
- changed `FlyToPathTrack` to be [complete] instead of [engine], as [complete] should probably be the codeword for "it exists and can be used, but it aint very useful"
- added `InputStartLeading` and `InputStopLeading` inputs

`func_lookdoor`:
- changed some fields from `string` to `float`

`trigger_playermovement`:
- elaborated on the "obsolete" spawnflag

`ai_script_conditions`:
- fixed the `NoValidActors` input

`env_credits`:
- added the `SetLogoLength` input

`env_fog`:
- added the `SetMaxDensity` input
- changed `SetMaxDensityLerpTo` to also include P2

`info_null`:
- added `SolidClass` version, as it works identically to mbase's `func_null`

`logic_choreographed_scene`:
- added `SetTarget#` inputs to tf2

`logic_eventlistener_itemequip`:
- fixed the `teamnum` keyvalue's choices

`logic_script`:
- fixed the whole conundrum with `group15`: valve wanted to add a `group16` in ASW, but instead renamed `group15` to it. TF2 also now has this bug.

`logic_timer`:
- changed some fields from `string` to `float`
- added the `UseRandomTime` input

`npc_enemyfinder`:
- changed the `fieldofview` keyvalue from `string` to `float`
- added the `FieldOfView`, `MinSearchDist`, and `MaxSearchDist` inputs

`npc_maker`:
- added the `relationship` keyvalue

`npc_sniper`:
- changed the `SetPaintInterval` and `SetPaintIntervalVariance` inputs to be episodic only

`phys_motor`:
- added the `addangaccel` keyvalue

`point_viewcontrol`:
- changed the `interpolatepositiontoplayer` keyvalue to be actually episodic only

`point_worldtext`:
- added it to TF2

`prop_glass_futbol_spawner`:
- added the `OnFutbolCaught` and `OnFutbolReleased` outputs: note that I have no clue if the descriptions are accurate, as I only noticed that VDC had those outputs but not HammerAddons

`tf_zombie`:
- added the `OnDeath` output

### CODE REVIEW

File: fgd/point/info/info_null.fgd [hunk 1]
```diff
@@ -9,3 +9,10 @@
 	[
 	@resources []
 	]
+
+// works identically to mbase's func_null when as a brush, so
+@SolidClass base(BaseEntity)
+= info_null: "A brush entity that's immediately removed upon spawning. Useful for invisible brush lights."
```
Comment (TeamSpen210) [inline fgd/point/info/info_null.fgd:15]: I’m not sure if having a `SolidClass` and `PointClass` with the same name will actually work, both my code and I think Valve’s will have one definition overwrite the other.

Comment (Adrthegamedev) [reply fgd/point/info/info_null.fgd:15]: hmm, good point. I was thinking of trying a hacky `BaseClass` to fix it, but the only use I can find for that to transfer properties (usually models) to entities...

Comment (TeamSpen210) [reply fgd/point/info/info_null.fgd:15]: I can do it in the postcompiler though, enable func_null if srctools is present, then swap the classname in the bsp.

Comment (Adrthegamedev) [reply fgd/point/info/info_null.fgd:15]: That could work :D

**Comment by Adrthegamedev** [comment]
sorry bout that; `worldtext()` didn't work how i thought it did

**Comment by TeamSpen210** [comment]
Not a problem. If you revert the `info_null` change also I think I can just merge this.

**Comment by Adrthegamedev** [comment]
aight, sounds good

**Comment by TeamSpen210** [comment]
Thanks very much!

**Pull request merged**